doc/fundamentals/getting_data_into_pachyderm.md

```diff
@@ -1,119 +1,92 @@
-
-
-Pachctl CLI
------------
-
-The pachctl CLI is the primary method of interaction with Pachyderm. To get data into Pachyderm, you should use the ``put-file`` command. Below are a example uses of ``put-file``. Go to :doc:`../pachctl/pachctl_put-file` for complete documentation. 
-
-.. note:: Commits in Pachyderm must be explicitly started and finished so ``put-file`` can only be called on an open commit (started, but not finished). The ``-c`` option allows you to start and finish the commit in addition to putting data as a one-line command. 
+```sh
+$ pachctl create-repo <repo name>
+```
 
+Then to put data into the created repo, you use the `put-file` command. Below are a few example uses of `put-file`, but you can see the complete documentation [here](../pachctl/pachctl_put-file). Note again, commits in Pachyderm must be explicitly started and finished so `put-file` can only be called on an open commit (started, but not finished). The `-c` option allows you to start and finish a commit in addition to putting data as a one-line command. 
 
 Add a single file:
 
-.. code-block:: shell
+```sh
+# first start a commit
+$ pachctl start-commit <repo>
 
- $ pachctl put-file <repo> <branch> -f <file>
+# then utilize the returned <commit-id> in the put-file request
+# to put <file> at <path> in the <repo>
+$ pachctl put-file <repo> <commit-id> <path> -f <file>
 
-Start and finish the commit while adding a file using ``-c``:
+# then finish the commit
+$ pachctl finish-commit <repo> <commit-id>
+```
 
-.. code-block:: shell
+Start and finish a commit while adding a file using `-c`:
 
- $ pachctl put-file -c <repo> <branch> -f <file> 
+```sh
+$ pachctl put-file <repo> <branch> <path> -c -f <file> 
+```
 
 Put data from a URL:
 
-.. code-block:: shell
-
- $ pachctl put-file <repo> <branch> -f http://url_path
-
-Add multiple files at once by using the ``-i`` option. The target file should be a list of files, paths, or URLs that you want to input all at once:
-
-.. code-block:: shell
-
- $ pachctl put-file <repo> <branch> -i <file>
-
-Pipe data from stdin into a file:
+```sh
+$ pachctl put-file <repo> <branch> <path> -c -f http://url_path
+```
 
-.. code-block:: shell
+Put data directly from an object store:
 
- $ echo "data" | pachctl put-file <repo> <branch> <path>
+```sh
+# here you can use s3://, gcs://, or as://
+$ pachctl put-file <repo> <branch> <path> -c -f s3://object_store_url
+``
 
-Add an entire directory by using the recursive flag, ``-r``:
+Add multiple files at once by using the `-i` option. The target file should be a list of files, paths, or URLs that you want to input all at once:
 
-.. code-block:: shell
+```sh
+$ pachctl put-file <repo> <branch> -c -i <file containing list of files, paths, or URLs>
+```
 
- $ pachctl put-file -r <repo> <branch> -f <dir>
+Pipe data from stdin into a data repository:
 
+```sh
+$ echo "data" | pachctl put-file <repo> <branch> <path> -c
+```
 
+Add an entire directory by using the recursive flag, `-r`:
```

Comment:
Great question.  As far as I know, it would be required if you are pointing to a single file in S3.  However, it would be required if you are pointing to a directory (I think, we should confirm).  Thus the question for `-r` is really whether you are pointing to a directory or not.  Again, we should confirm.